src/odemis/driver/technolution.py

```diff
@@ -0,0 +1,544 @@
+import copy
+import json
+import queue
+from datetime import datetime
+import logging
+import time
+import threading
+import requests
+import numpy
+from odemis import model
+import weakref
+
+from odemis import model
+from src.openapi_server.models.mega_field_meta_data import MegaFieldMetaData
+from src.openapi_server.models.cell_parameters import CellParameters as CellAcqParameters
+from src.openapi_server.models.field_meta_data import FieldMetaData
+from src.openapi_server.models.error_status import ErrorStatus
+
+from odemis.model import _vattributes
+
+_METHOD_GET = 1
+_METHOD_POST = 2
+
+
+class AcquisitionServer(model.HwComponent):
+
+    def __init__(self, name, role, server_URL, children={}, deamon=None, **kwargs):
+        super(AcquisitionServer, self).__init__(name, role, **kwargs)
+
+        self.server_URL = server_URL
+        # TODO K.K. Set external storage and connection
+
+        # Stop any acquisition if already one was in progress
+        self.simpleASMCall(self.server_URL + "/scan/finish_mega_field", _METHOD_POST, 204)
+
+        # TODO K.K. remove *kwargs and include setup of entire component in try (each sperated) first check if childeren
+        # holds value of component
+        try:
+            kwargs = children["MirrorDescanner"]
+        except Exception:
+            raise ValueError("Required child MirrorDescanner not provided")
+        self._MirrorDescanner = MirrorDescanner("MirrorDescanner", role=None, parent=self)
```

Comment:
Do you mean each component should have it's own keyword in the ASM _ _init__ for the role/name?
So that the ASM is called by:
ASM(name, role, mirror_name='foo', mirror_role='bar', mppc_name='mppc' .....
or should all the components have the same name?

Reply:
I now understand how to do this, implemented!